code/src/index.css

```diff
@@ -5,9 +5,126 @@ body {
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background-color: black;
 }
 
 code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
+/* MOVIE LIST */
+
+.movie-list {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: center;
+  word-wrap: break-word;
+}
+
+/* MOVIE CARD */
+
+.movie-card-overlay {
+  position: relative;
+}
+
+.text-overlay {
+  position: absolute;
+  color:#fff;
+  background: rgba(0, 0, 0, 0.45);
+  width: 100%;
+  height: 100%;
+  opacity: 0;
+  transition: 0.4s ease;
+  bottom: 0;
+  top: 0;
+  left:0;
+  right:0;
+}
+
+.text-overlay h1 {
+  position: absolute;
+  bottom: 50px;
+  color:#fff;
+  font-size: 2em;
+  font-weight: bold;
+  text-decoration: none;
+  margin:20px 0 10px 15px;
+}
+
+.text-overlay p {
+  position: absolute;
+  bottom: 0;
+  color: #fff;
+  font-size: 15px;
+  text-decoration: none;
+  margin: 5px 0 20px 15px
+}
+
+.text-overlay:hover {
+  opacity: 1;
+}
+
+/* MOVIE PAGE */
+
+.movie-bg-img {
+  min-height: 100vh;
+  display: flex;
+  justify-content: flex-end;
+  background-size: cover;
+}
+
+.movie-page-box {
+  display: flex;
+  align-items: flex-end;
+  width: 100%;
+  padding: 50px;
+}
+
+.movie-page-poster {
+  border: solid #fff 3px;
+}
+
+.movie-page-text {
+  color: #fff;
+  max-width: 450px;
+  margin-left: 20px;
+}
+
+.rating {
+  font-size: 17px;
+  color: #FEB12F;
+  font-weight: 400;
+  margin-left: 10px;
+}
+
+/* BUTTON BACK */
+
+.button {
+  display: flex;
+  justify-content: center;
+  align-items:center;
+  position: absolute;
+  top: 40px;
+  left: 40px;  
+  padding: 8px;
+}
+
+.arrow-back {
+  width: 40px;
+  height: 40px;
+}
+
+.button-text {
+  font-size: 20px;
+  font-weight: 700;
+  color: #fff;
+  margin: 0 2px;
+}
+
+.button-text:hover {
+  margin-left: 7px;
+  transition: 0.3s ease-in-out;
```

Comment:
The transition should be on the original class element (so not the :hover). It works OK but it does not ease-in-out, only ease out. So perhaps moving it will fix this. For example on your .text-overlay you have done this correctly!